File: Data Structures & Algorithms/k-closest-points-to-origin/submission-0.cpp
class Solution {
public:

    struct Point {
        int x, y;
        Point(int x, int y){
            this->x = x;
            this->y = y;
        }

        bool operator < (const Point &other) const {
            return x*x + y*y > other.x * other.x + other.y * other.y;
        }
    };

    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        priority_queue<Point> pq;
        for(vector<int> coord : points){
            pq.push(Point(coord[0], coord[1]));
        }

        vector<vector<int>> ans;
        
        while(k--){
            vector<int> coord;
            Point closestPoint = pq.top();
            pq.pop();
            coord.push_back(closestPoint.x);
            coord.push_back(closestPoint.y);

            ans.push_back(coord);
        }

        return ans;
    }
};

File: Data Structures & Algorithms/combinations/submission-0.py
class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        ans = []
        run = []
        def dfs(i):
            if len(run) == k:
                ans.append(run[:])
                return
            
            for j in range(i, n + 1):
                run.append(j)
                dfs(j + 1)
                run.pop()
        
        dfs(1)
            
        return ans